packages/camera/camera_windows/windows/capture_controller.cpp

```diff
@@ -860,9 +875,31 @@ void CaptureControllerImpl::OnRecordStopped(CameraResult result,
 // Implements CaptureEngineObserver::UpdateBuffer.
 bool CaptureControllerImpl::UpdateBuffer(uint8_t* buffer,
                                          uint32_t data_length) {
+  using flutter::EncodableValue, flutter::EncodableMap;
+
   if (!texture_handler_) {
     return false;
   }
+  if (image_stream_sink_) {
+    // Encode the frame into a map that can be sent through the `EventSink`.
+    EncodableMap encoded_frame = {
+        {EncodableValue("data"),
+         EncodableValue(std::vector<uint8_t>(buffer, buffer + data_length))},
+        {EncodableValue("height"),
+         EncodableValue(static_cast<int64_t>(preview_frame_height_))},
+        {EncodableValue("width"),
+         EncodableValue(static_cast<int64_t>(preview_frame_width_))},
+        {EncodableValue("length"),
+         EncodableValue(static_cast<int64_t>(data_length))}};
```

Comment:
Rather than using raw method channels, can you use `CameraApi::GetCodec` as the codec, and use a Pigeon-defined class here, to get strong typing?

(We've done something very similar with plugins that use platform view arguments, where we define a class in Pigeon that's not used in the Pigeon-managed APIs, and then use that class in other places where things are serialized via channel codecs.)

Reply:
Can you point to an example where you’ve done this? It seems like I’d have to switch to using a `BasicMessageChannel` to take advantage of `CameraApi::GetCodec`.

Reply:
> Can you point to an example where you’ve done this?

There aren't any C++ examples, so I don't think they would be very helpful.

> It seems like I’d have to switch to using a `BasicMessageChannel` to take advantage of `CameraApi::GetCodec`.

Why? The codec is [passed when creating the event channel](https://github.com/flutter/packages/pull/8234/files#diff-97976b5de638b37a30e4e107979a26a59358a211c0ac72944f9f053c6eb9b763R369).